src/providers/database.ts

```diff
@@ -206,11 +206,10 @@ export class DeltaSnapshot implements firebase.database.DataSnapshot {
     return valAt(this._delta, this._childPath) !== undefined;
   }
 
-  // TODO(inlined) what is this boolean for?
   forEach(action: (a: DeltaSnapshot) => boolean): boolean {
     let val = this.val();
     if (_.isPlainObject(val)) {
-      _.keys(val).forEach(key => action(this.child(key)));
+      return _.keys(val).some(key => action(this.child(key)));
```

Comment:
Also, I think we want to make sure this behavior only happens when `true` is returned, not when a truthy value (such as `"a"` or `1`) is returned. So, this should probably end with `action(this.child(key) === true`. Can you please add a test for the case where `action()` returns `1` to ensure this behavior is correct?